Feature/pratice2

kwakdaeyoung opened the pull request:
너무 늦었지만 재미있었습니다 ^^

kwakdaeyoung committed:
createYS/midasit_tgc_wdb_dev:dykwak

creativeYS committed:
practice02

creativeYS committed:
practice02

creativeYS committed:
practice02
- Component, Hooks example
- update readme.md

kwakdaeyoung committed:
practice02 과제
- Component, Hooks example
- update readme.md

creativeYS commented on practice02/src/components/Todos.js:
```diff
@@ -0,0 +1,80 @@
+        {unfinishedItems.map((todo) => (
+          <li key={todo.id} className={`unfinished ${todo.completed ? 'completed' : ''}`}>
+            {todo.todo}
+          </li>
```
이부분은 그냥 완료되지 않은 항목들만 추려서 달랑달랑 표현해주는건가요? ㅎㅎㅎㅎ

kwakdaeyoung replied:
넵 에니메이션을 해보고 싶어서 해봤습니다 
GPT의 도움을 받아서요 ㅎ

creativeYS commented on practice02/src/components/Todos.js:
```diff
@@ -0,0 +1,80 @@
+    // 일정 시간 후에 unfinishedItems의 클래스를 toggle하여 애니메이션을 활성화합니다.
+    const timeout = setTimeout(() => {
+      setUnfinishedItems([]);
+    }, 3000); // 3초 후에 애니메이션을 시작합니다.
```
이것은 3초후에 에니메이션을 종료(언피니시드 아이템을 삭제) 하는것이죠?

kwakdaeyoung replied:
넵 3초 후에 종료하는 것입니다 
주석이 이상하네요 ㅎㅎ GPT가 달아줬네요

creativeYS commented on practice02/src/components/Todos.js:
```diff
@@ -0,0 +1,80 @@
+      {/* 항목 개수 정보 표시 */}
+      <p className="item-info-left">전체 항목 수: {todos.length}</p>
+      <p className="item-info-center">완료된 항목 수: {completedCount}</p>
+      <p className="item-info-right">완료되지 않은 항목 수: {notCompletedCount}</p>
```
스타일을 className으로 css를 활용해서 잘 적용해주셨습니다!

kwakdaeyoung replied:
ㅎㅎ감사합니다

creativeYS commented on practice02/src/components/Todos.js:
```diff
@@ -0,0 +1,80 @@
+          todos.map((todo) => (
+            <li key={todo.id} className={todo.completed ? 'completed' : ''}>
+              {todo.todo}
+            </li>
```
리스트 항목들을 별도 컴포넌트로 분리해보면 어떨까요?
하나의 컴포넌트에 모든 요소를 몰아넣기보다 적절한 규모로 분리하면 사용하기 좋을 것 같스빈다.

kwakdaeyoung replied:
넵 한번 해보겠습니다 ^^
코드리뷰 감사합니다 ~

kwakdaeyoung committed:
todo.css 파일 추가

# Conflicts:
#	practice02/src/App.js
#	practice02/src/components/Todos.css
#	practice02/src/components/Todos.js

kwakdaeyoung committed:
체크박스 상태 변경 반영되도록 수정

kwakdaeyoung commented:
같은 브랜치이름으로 해서 다 덮어쓰고 다시 올릴게요

kwakdaeyoung commented:
알려주신대로 css 파일 만들어서 올립니다
감사합니다

kwakdaeyoung committed:
필요없는 코드 제거

creativeYS commented:
피드백 반영까지 해주셨군요! 감사합니다. 병합하겠습니다
